modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -83,115 +82,161 @@ def _read(cls, filepath_or_buffer, **kwargs):
-            # Launch tasks to read partitions
-            partition_ids = []
-            index_ids = []
-            dtypes_ids = []
-            # Max number of partitions available
-            num_partitions = NPartitions.get()
-            # This is the number of splits for the columns
-            num_splits = min(len(column_names), num_partitions)
-            # Metadata
-            column_chunksize = compute_chunksize(empty_pd_df, num_splits, axis=1)
-            if column_chunksize > len(column_names):
-                column_widths = [len(column_names)]
-                # This prevents us from unnecessarily serializing a bunch of empty
-                # objects.
-                num_splits = 1
-            else:
-                column_widths = [
-                    column_chunksize
-                    if len(column_names) > (column_chunksize * (i + 1))
-                    else 0
-                    if len(column_names) < (column_chunksize * i)
-                    else len(column_names) - (column_chunksize * i)
-                    for i in range(num_splits)
-                ]
-
-            args = {
-                "fname": filepath_or_buffer,
-                "num_splits": num_splits,
-                **partition_kwargs,
-            }
 
+        with cls.file_open(filepath_or_buffer_md, "rb", compression_infered) as f:
             splits = cls.partitioned_file(
                 f,
                 num_partitions=num_partitions,
-                nrows=nrows,
+                nrows=kwargs.get("nrows", None),
                 skiprows=skiprows,
                 quotechar=quotechar,
                 is_quoting=is_quoting,
             )
-            for start, end in splits:
-                args.update({"start": start, "end": end})
-                partition_id = cls.deploy(cls.parse, num_splits + 2, args)
-                partition_ids.append(partition_id[:-2])
-                index_ids.append(partition_id[-2])
-                dtypes_ids.append(partition_id[-1])
-
-        # Compute the index based on a sum of the lengths of each partition (by default)
-        # or based on the column(s) that were requested.
+
+        partition_ids, index_ids, dtypes_ids = cls._launch_tasks(
+            splits, **partition_kwargs
+        )
+
+        new_query_compiler = cls._get_new_qc(
+            partition_ids=partition_ids,
+            index_ids=index_ids,
+            dtypes_ids=dtypes_ids,
+            index_col_md=index_col,
+            index_name=empty_pd_df.index.name,
+            column_widths=column_widths,
+            column_names=column_names,
+            squeeze=kwargs.get("squeeze", False),
+            skipfooter=kwargs.get("skipfooter", None),
+            parse_dates=kwargs.get("parse_dates", False),
+        )
+        return new_query_compiler
+
+    # _read helper functions
+    @classmethod
+    def _read_csv_check_support(
+        cls, filepath_or_buffer, read_csv_kwargs, compression_infered
```

Comment:
Could we add type hinting for this or are the arguments too tedious to add type hints for (if this is the case, my bet would be that `read_csv_kwargs` might be challenging?)? 

If `read_csv_kwargs` is too hard to had type hints for, would it affect readability if we only passed in the arguments needed? I think we only need a couple arguments from `read_csv_kwargs`

Reply:
that looks good!